Base/Python/slicer/util.py

```diff
@@ -639,17 +650,17 @@ def reloadScriptedModule(moduleName):
     except AttributeError:
       pass
 
-  # remove spacer items
+  # call cleanup function for the existing widget
+  if hasattr(slicer.modules, widgetName):
+    w = getattr(slicer.modules, widgetName)
+    w.cleanup()
```

Comment:
Is cleanup called again when module is about to be unloaded?
Should we call something like this to avoid double cleanup?

    slicer.app.moduleManager().disconnect('moduleAboutToBeUnloaded(QString)', w._onModuleAboutToBeUnloaded)

Reply:
Good point, the signal should indeed be disconnected.